velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h

```diff
@@ -179,7 +179,7 @@ class InputStreamFileSystem : public ::duckdb::FileSystem {
   }
 
  private:
-  int nextStreamId_ = 1;
+  std::atomic<int> nextStreamId_ = 1;
```

Comment:
While working on this PR, I occasionally noticed the following core. I was able to reproduce the core in a 500 run test. With this fix, I do not see this issue anymore.
CI run that also hit this core
https://circleci.com/api/v1.1/project/github/facebookincubator/velox/22031/output/109/0?file=true&allocation-id=61f4846f768d232005f83141-0-build%2F3073EF21
```
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0129 00:26:54.578812  9820 Exceptions.h:68] ../.././velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h46OpenFileit != lockedStreams->end()Unknown stream with ID 12RUNTIMEINVALID_STATE
E0129 00:26:54.579144  9820 Exceptions.h:68] ../.././velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h63CloseStreamit != lockedStreams->end()Unknown stream with ID 12RUNTIMEINVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
unknown file: Failure
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Unknown stream with ID 12
Retriable: False
Expression: it != lockedStreams->end()
Function: OpenFile
File: ../.././velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h
Line: 46
Stack trace:
```

Reply:
@majetideepak Thank you for debugging and fixing. It would be nice to mention this fix in the PR description.